transpiler/src/ast/core/BitloopsVisitor/helpers/setup/corsOptionsVisitor.ts

```diff
@@ -0,0 +1,54 @@
+import { BitloopsTypesMapping } from '../../../../../helpers/mappings.js';
+import BitloopsParser from '../../../../../parser/core/grammar/BitloopsParser.js';
+import { CorsOptionsNodeBuilder } from '../../../intermediate-ast/builders/setup/CorsOptionsNodeBuilder.js';
+import { CorsOriginNodeBuilder } from '../../../intermediate-ast/builders/setup/CorsOriginNodeBuilder.js';
+import { CorsOptionsListNode } from '../../../intermediate-ast/nodes/setup/CorsOptionsListNode.js';
+import { CorsOptionsNode } from '../../../intermediate-ast/nodes/setup/CorsOptionsNode.js';
+import { CorsOriginNode } from '../../../intermediate-ast/nodes/setup/CorsOriginNode.js';
+import BitloopsVisitor from '../../BitloopsVisitor.js';
+import { produceMetadata } from '../../metadata.js';
+import { stringEvaluation } from '../index.js';
+
+export const corsOptionsListStatementVisitor = (
+  thisVisitor: BitloopsVisitor,
+  ctx: BitloopsParser.CorsOptionsListStatementContext,
+): CorsOptionsNode => {
+  const metadata = produceMetadata(ctx, thisVisitor);
+
+  const corsOptionsListNode: CorsOptionsListNode[] = thisVisitor.visit(ctx.corsOptionsList());
+
+  const corsOriginNode: CorsOriginNode = corsOptionsListNode.find(
+    (node) => node.getNodeType() === BitloopsTypesMapping.TCorsOrigin,
+  );
```

Comment:
Why is this needed? Shouldn't the whole `corsOptionsListNode` be passed to `CorsOptionsNodeBuilder`. If it is needed maybe this logic should be moved to a method inside `CorsOptionsListNode`.